src/irr/asset/CImageWriterJPG.cpp

```diff
@@ -98,16 +96,82 @@ static void jpeg_file_dest(j_compress_ptr cinfo, io::IWriteFile* file)
 	dest->file = file;
 }
 
+template<asset::E_FORMAT outFormat>
+core::smart_refctd_ptr<asset::ICPUImage> getJPGConvertedOutput(const asset::ICPUImage* image)
+{
+	using CONVERSION_FILTER = asset::CConvertFormatImageFilter<asset::EF_UNKNOWN, outFormat>;
+
+	core::smart_refctd_ptr<asset::ICPUImage> newConvertedImage;
+	{
+		auto referenceImageParams = image->getCreationParameters();
+		auto referenceBuffer = image->getBuffer();
+		auto referenceRegions = image->getRegions();
+		auto referenceRegion = referenceRegions.begin();
+		const auto newTexelOrBlockByteSize = asset::getTexelOrBlockBytesize(outFormat);
+
+		asset::TexelBlockInfo referenceBlockInfo(referenceImageParams.format);
+		core::vector3du32_SIMD referenceTrueExtent = referenceBlockInfo.convertTexelsToBlocks(referenceRegion->getTexelStrides());
+
+		auto newImageParams = referenceImageParams;
+		auto newCpuBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(referenceTrueExtent.X * referenceTrueExtent.Y * referenceTrueExtent.Z * newTexelOrBlockByteSize);
+		auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1);
+		newRegions->front() = *referenceRegion;
```

Comment:
no you want to define a whole new region, full 2D extent (depth==1), mipLevel 0, base array level 0 and array count 1

Reply:
haven't you seen it? https://github.com/buildaworldnet/IrrlichtBAW/blob/2e41fbce4dde797df22532800d6686bb1fec86a8/src/irr/asset/CImageWriterJPG.cpp#L257

Reply:
that's why I asked you recently how to get entire single layer, only one specified by the user

Reply:
so why make this look like a generalpurpose function if its only intended to work with an image created with `IImageAssetHandlerBase::getTopImageDataForCommonWriting` ?

Reply:
image at that point is translated to have only top data, the writer encapsulates it's implementation, I don't find it much wrong  solution

Reply:
but I don't undestand quite what you mean

Reply:
to be honest there's no reason for `IImageAssetHandlerBase::getTopImageDataForCommonWriting` to exist, if you have a function like this (it could just take an image view as an input instead of image)